python/tank/descriptor/io_descriptor/factory.py

```diff
@@ -109,6 +110,10 @@ def create_io_descriptor(
 
     # specify where to go look for caches
     descriptor.set_cache_roots(bundle_cache_root, fallback_roots)
+    # also specify whether this is using a non-default bundle cache
+    # this will be used further down the path to determine whether or not
+    # the content is purgeable.
+    descriptor.set_use_non_default_bundle_cache_root(use_non_default_bundle_cache)
```

Comment:
make this a property, e.g.

```
descriptor.uses_non_default_primary_bundle_cache_root = use_non_default_bundle_cache
```

Reply:
The rational was that there was another related method, I simply copied style from there.
```
    # specify where to go look for caches
    descriptor.set_cache_roots(bundle_cache_root, fallback_roots)
    # also specify whether this is using a non-default bundle cache
    # this will be used further down the path to determine whether or not
    # the content is purgeable.
    descriptor.set_use_non_default_bundle_cache_root(
        bundle_cache_root != _get_default_bundle_cache_root()
    )
```